snapcraft/plugins/nodejs.py

```diff
@@ -85,8 +94,11 @@ def build(self):
         self._nodejs_tar.provision(self.installdir)
         for pkg in self.options.node_packages:
             self.run(['npm', 'install', '-g', pkg])
-        if os.path.exists(os.path.join(self.builddir, 'package.json')):
-            self.run(['npm', 'install', '-g'])
+        if os.path.exists(os.path.join(self.builddir,
+                                       self.options.node_folder,
+                                       'package.json')):
+            self.run(['npm', 'install', '-g',
+                     os.path.join(self.builddir, self.options.node_folder)])
```

Comment:
OK,

I think its correct this way, but i now see that its smarter to use
the default '' rather than ''. In that way the dir of join will be
exactly the same as without this patch for the current parts that dont
specify node-folder... let me change that and then it should safe...

Wait for resubmit!

 - Alexander


2015-12-24 15:59 GMT+01:00 Alexander Sack <asacasa@gmail.com>:
> 2015-12-23 19:14 GMT+01:00 Kyle Fazzari <notifications@github.com>:
>>
>> In snapcraft/plugins/nodejs.py:
>>
>> > @@ -85,8 +94,11 @@ def build(self):
>> >          self._nodejs_tar.provision(self.installdir)
>> >          for pkg in self.options.node_packages:
>> >              self.run(['npm', 'install', '-g', pkg])
>> > -        if os.path.exists(os.path.join(self.builddir, 'package.json')):
>> > -            self.run(['npm', 'install', '-g'])
>> > +        if os.path.exists(os.path.join(self.builddir,
>> > +                                       self.options.node_folder,
>> > +                                       'package.json')):
>> > +            self.run(['npm', 'install', '-g',
>> > +                     os.path.join(self.builddir, self.options.node_folder)])
>>
>> Just to verify (not a nodejs expert here): it's safe to do this even when the package.json IS in the root of the project?
>
> node.path definition api doc says:
>
> ```
> Note: If the arguments to join have zero-length strings, unlike other
> path module functions, they will be ignored. If the joined path string
> is a zero-length string then '.'will be returned, which represents the
> current working directory.
> ``` - https://nodejs.org/api/path.html#path_path_join_path1_path2
>
> so if node_folder is '' (zero-length) or '.' the result will be same
> as $builddir and $builddir/. always poing to same directory as this
> one... or just builddir, which should be fine ...
>
>> path.join ("asdda", "subdir", "package.json");
> 'asdda/subdir/package.json'
>> path.join ("asdda", "", "package.json");
> 'asdda/package.json'
>> path.join ("asdda", ".", "package.json");
> 'asdda/package.json'
>
>
>> —
>> Reply to this email directly or view it on GitHub.